File: 3289-the-two-sneaky-numbers-of-digitville/3289-the-two-sneaky-numbers-of-digitville.cpp
class Solution {
public:
    vector<int> getSneakyNumbers(vector<int>& nums) {
        vector<int> res;
        unordered_map<int, int> countMap;
        for (int num : nums) countMap[num]++;

        // Debug: Print the countMap
        cout << "CountMap:" << endl;
        for (const auto& [num, count] : countMap) {
            cout << "Number: " << num << ", Count: " << count << endl;
        }   

        for (const auto& [num, count] : countMap) {
            if (count > 1) res.push_back(num);
        }

        return res;     
    }
};

File: 3289-the-two-sneaky-numbers-of-digitville/3289-the-two-sneaky-numbers-of-digitville.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var getSneakyNumbers = function(nums) {
    tally = {}

    nums.forEach(num => tally[num] = (tally[num] || 0) + 1 );

    return Object.keys(tally)
        .filter(key => tally[key] > 1)
        .map(Number); // Convert keys back to numbers    
};